domain/status/storage.go

```diff
@@ -0,0 +1,137 @@
+// Copyright 2025 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package status
```

Comment:
What is the justification for special casing storage in this instance to a seperate file?

Reply:
To avoid shoving everything in one huge file. There will be several storage methods by the time we're done. A similar thing has been done in the application domain for units, expose, charm etc